streams/cluster/src/test/java/com/hortonworks/streamline/streams/cluster/register/impl/KafkaServiceRegistrarTest.java

```diff
@@ -51,47 +54,70 @@ public void testRegister() throws Exception {
 
         KafkaServiceRegistrar registrar = initializeServiceRegistrar();
 
-        try (InputStream is = getClass().getClassLoader().getResourceAsStream(SERVER_PROPERTIES_FILE_PATH)) {
-            Config config = new Config();
-            config.put(KafkaServiceRegistrar.PARAM_KAFKA_BROKER_HOSTNAMES, Lists.newArrayList("kafka-1","kafka-2"));
-            ManualServiceRegistrar.ConfigFileInfo serverProperties = new ManualServiceRegistrar.ConfigFileInfo(SERVER_PROPERTIES, is);
-            registrar.register(cluster, config, Lists.newArrayList(serverProperties));
-        }
+        Config config = new Config();
+        config.put(KafkaServiceRegistrar.PARAM_ZOOKEEPER_CONNECT, "zookeeper-1:2181,zookeeper-2:2181,zookeeper-3:2181");
+        config.put(KafkaServiceRegistrar.PARAM_LISTENERS, "SASL_PLAINTEXT://kafka-1:6668,PLAINTEXT://kafka-2:6669,SSL://kafka-3:6670,SASL_SSL://kafka-4:6671");
+        config.put(KafkaServiceRegistrar.PARAM_SECURITY_INTER_BROKER_PROTOCOL, "SSL");
+        registrar.register(cluster, config, Collections.emptyList());
 
         Service kafkaService = environmentService.getServiceByName(cluster.getId(), Constants.Kafka.SERVICE_NAME);
         assertNotNull(kafkaService);
+
+        Component broker = environmentService.getComponentByName(kafkaService.getId(), ComponentPropertyPattern.KAFKA_BROKER.name());
+        assertNotNull(broker);
+
+        Collection<ComponentProcess> brokerProcesses = environmentService.listComponentProcessesInComponent(broker.getId());
+        assertEquals(4, brokerProcesses.size());
```

Comment:
why 4? Is it NUM_LISTENERS?  can you put a constant describing what this number is?